Metadata/MetadataConverter.php

```diff
@@ -64,6 +64,14 @@ public function convert(ClassHierarchyMetadata $metadata)
             $definition->setTags($classMetadata->tags);
             $definition->setProperties($classMetadata->properties);
 
+            if (null !== $classMetadata->decorates && method_exists($definition, 'setDecoratedService')) {
+                $definition->setDecoratedService($classMetadata->decorates, $classMetadata->decoration_inner_name);
```

Comment:
wouldn't do that - in a bundle I add the deprecation / decoration annotation, the bundle might work with symfony 2.3, and is now included in a project using 2.3 -> would throw the exception then

both decorating and deprecation support was added in 2.8